### DISCUSSION TITLE
Multi row Selection Expanding

### DISCUSSION THREAD

**Discussion by JBlaz** [BetterThanTomorrow/calva#1218] [Ideas]
If I want to remove something on multiple rows I cannot select multiple rows

          [weekday-circle {:weekday 0 :disabled @weekdays-disabled?} "Su"]
          [weekday-circle {:weekday 1 :disabled @weekdays-disabled?} "Mo"]
          [weekday-circle {:weekday 2 :disabled @weekdays-disabled?} "Tu"]
          [weekday-circle {:weekday 3 :disabled @weekdays-disabled?} "We"]
          [weekday-circle {:weekday 4 :disabled @weekdays-disabled?} "Th"]
          [weekday-circle {:weekday 5 :disabled @weekdays-disabled?} "Fr"]
          [weekday-circle {:weekday 6 :disabled @weekdays-disabled?} "Sa"]

If I put the cursor before the @ symbol on all lines then ctrl+shift+right arrow it only selects the first row. Any ideas how to fix this?

**Comment by PEZ** [comment]
Hello! Paredit doesn't currently support multi cursors. See https://github.com/BetterThanTomorrow/calva/issues/610

In this particular case you can probably use the VS Code built in *Expand Selection* command. On Windows and Linux Calva overrides its keyboard binding, so you either need to make the two commands use different keybindings, or use the built-in command from the Command Palette.

**Reply by JBlaz** [reply]
Thanks so much for the response. I would very much like this to be a feature any suggestions on where to start thinking of a solution?

**Reply by JBlaz** [reply]
Meaning line number in the code or file which contains logic for the already implemented Selection Expanding?

**Reply by PEZ** [reply]
Super happy you ask!

I think it goes deeper than grow selection, but maybe it's a good start to first figure that use case out. Calva Paredit has a selections tack where it pushes and pops selections. Growing selection is a matter of first finding out what an expanded selection looks like, then pushing it on the stack. And, of course, actually selecting that. Shrinking the selection is just a matter of popping this stack + selecting whatever is below what was popped.

Here is where the figuring out and pushing on the stack happens: https://github.com/BetterThanTomorrow/calva/blob/157a8cfd7e8fb5b8ce91301027e482dcfeace01c/src/cursor-doc/paredit.ts#L558

**Reply by JBlaz** [reply]
Many thanks! A simple work around, and maybe what should be done in the mean time is to change the when clause of the short cut to exclude multi line selections. I changed the when case of `Calva Paredit: Select Backward Sexp` and `Calva Paredit: Select Forward Sexp` to this: `calva:keybindingsEnabled && editorTextFocus && !editorHasMultipleSelections && editorLangId == 'clojure' && editorLangId == 'clojure' && paredit:keyMap =~ /original|strict/` note the addition of`!editorHasMultipleSelections`

**Reply by PEZ** [reply]
I'd be happy to merge a PR that adds this workaround to https://calva.io. Not sure about including the change in Calva. It won't add to Calva's selection stack which sort of breaks the contract...

**Reply by JBlaz** [reply]
That last statement is sort of contradictory to me. I would be happy to create a PR but you said add it to calva but not sure about adding it calva?

**Reply by JBlaz** [reply]
I created a PR https://github.com/BetterThanTomorrow/calva/pull/1220. I hope I did it right.

**Reply by PEZ** [reply]
I meant adding info about the workaround to the documentation. But I'll have a think about wether we should change Calva's defaults.

**Reply by JBlaz** [reply]
Gotcha.
